README.md

```diff
@@ -1,7 +1,103 @@
+
+
+A simple request for an addon would look like this:
+```bash
+host="${HOST:-localhost:8080/hub}"
+
+# Make a request to hub
+request_cmd="$(curl -i -o - -X POST ${host}/tasks -d \
+'{
+    "name":"Jkube",
+    "state": "Ready",
+    "locator": "jkube",
+    "addon": "jkube",
+    "application": {"id": 2},
+    "data": {}
+}')"
+```
+
+In above request we mention `addon: jkube` as an addon identifier and `application: {id: 2}` states the application against which this addon will be executed. The `data` field here is empty but we can pass data to addon during api request and we use following struct to receive data for the addon.
+
+```go
+// Data Addon data passed in the secret.
+type Data struct {
+	// Output directory within application bucket.
+	Output string `json:"output" binding:"required"`
+	// Mode options.
+	Mode Mode `json:"mode"`
+	// Sources list.
+	Sources Sources `json:"sources"`
+	// Targets list.
+	Targets Targets `json:"targets"`
+	// Scope options.
+	Scope Scope `json:"scope"`
+	// Rules options.
+	Rules *Rules `json:"rules"`
+}
+```
+
+Once a request is received - addon will fetch the application mentioned in the request and perform the operations defined in the addon. For this addon we perform following operations:
+- Fetch the application (clone github/subversion repository)
+- Parse the maven config file and add kubernetes-maven-plugin to `pom.xml`
+- Build the application using maven build
+- Use kubernetes-maven-plugin to generate container image and k8s manifests
+- Commit the generated resources to repo
+
+## Development Environment Setup
+
+Steps to successfully execute Jkube addon:
+- Setup Minikube and install tackle - [start-minikube action](https://github.com/konveyor/tackle2-operator/tree/main/.github/actions/start-minikube), [install-tackle action](https://github.com/konveyor/tackle2-operator/tree/main/.github/actions/install-tackle), [install-tackle](https://github.com/konveyor/tackle2-operator/blob/main/hack/install-tackle.sh)
+
+- Forward port for tackle-ui service
+  ```bash
+  kubectl port-forward service/tackle-ui 8080:8080 -n konveyor-tackle
+  ```
+- Populate required data in tackle, including application against which addon will be executed - [reference](https://github.com/konveyor/tackle2-hub/tree/main/hack/add), [jkube-data](https://github.com/konveyor/tackle2-addon-jkube/blob/main/hack/task.sh)
+
+- Add Jkube addon to tackle - [jkube addon](https://github.com/konveyor/tackle2-addon-jkube/blob/main/hack/addon.yml)
+  ```bash
+  kubectly apply -f addon.yml
+  ```
+- Make curl request to Jkube addon - [request]()
+  ```bash
+  host="${HOST:-localhost:8080/hub}"
+
+  # Make a request to hub
+  request_cmd="$(curl -i -o - -X POST ${host}/tasks -d \
+  '{
+      "name":"Jkube",
+      "state": "Ready",
+      "locator": "jkube",
+      "addon": "jkube",
+      "application": {"id": 1},
```

Comment:
You'll need to tell people how to create an application, either through the UI or command line.

Reply:
I see. Maybe better to be more explicit? ie. Create an application in tackle with this command (and that application is one jkube addon will work correctly). We should also explain so long as we are using `curl` how to get the application's id for use here.